app/models/user.rb

```diff
@@ -105,12 +107,24 @@ def session_active?(id)
     session_activations.active? id
   end
 
+  def ldap_user?
+    Rails.configuration.x.use_ldap && ldap_dn.present?
+  end
+
   protected
 
   def send_devise_notification(notification, *args)
     devise_mailer.send(notification, self, *args).deliver_later
   end
 
+  def confirmation_required?
+    if ldap_user?
+      !Rails.configuration.x.ldap_skip_email_confirmation
```

Comment:
uh nope, it would be `ldap_user? && !Rails.configuration.x.ldap_skip_email_confirmation || super`, but then it's much harder to read?